Test/Classes/TestTextUtils.cs

```diff
@@ -37,12 +37,102 @@ public void StripDateFromNameFormats()
+
+            Assert.Equal("Test", TextUtils.StripDateFromName("Test 3-2-09", date));
+            Assert.Equal("Test", TextUtils.StripDateFromName("Test 3-2-2009", date));
+            Assert.Equal("Test", TextUtils.StripDateFromName("Test 3-02-09", date));
+            Assert.Equal("Test", TextUtils.StripDateFromName("Test 3-02-2009", date));
+
+            Assert.Equal("Test", TextUtils.StripDateFromName("Test 3/2/09", date));
+            Assert.Equal("Test", TextUtils.StripDateFromName("Test 3/2/2009", date));
+            Assert.Equal("Test", TextUtils.StripDateFromName("Test 3/02/09", date));
+            Assert.Equal("Test", TextUtils.StripDateFromName("Test 3/02/2009", date));
+
+            Assert.Equal("Test", TextUtils.StripDateFromName("Test 3 Feb 09", date));
+            Assert.Equal("Test", TextUtils.StripDateFromName("Test 3 Feb '09", date));
+            Assert.Equal("Test", TextUtils.StripDateFromName("Test 3 Feb 2009", date));
+
+            Assert.Equal("Test", TextUtils.StripDateFromName("Test 3-Feb-09", date));
+            Assert.Equal("Test", TextUtils.StripDateFromName("Test 3-Feb-2009", date));
+
+            Assert.Equal("Test", TextUtils.StripDateFromName("Test 3 February 09", date));
+            Assert.Equal("Test", TextUtils.StripDateFromName("Test 3 February '09", date));
+            Assert.Equal("Test", TextUtils.StripDateFromName("Test 3 February 2009", date));
+
+            Assert.Equal("Test", TextUtils.StripDateFromName("Test 3rd Feb 2009", date));
+            Assert.Equal("Test", TextUtils.StripDateFromName("Test 3rd February 2009", date));
+
+            Assert.Equal("Test", TextUtils.StripDateFromName("Test 2009-2-3", date));
+            Assert.Equal("Test", TextUtils.StripDateFromName("Test 2009/2/3", date));
+            Assert.Equal("Test", TextUtils.StripDateFromName("Test 2009.2.3", date));
+            Assert.Equal("Test", TextUtils.StripDateFromName("Test 2009-2-03", date));
+            Assert.Equal("Test", TextUtils.StripDateFromName("Test 2009/2/03", date));
+            Assert.Equal("Test", TextUtils.StripDateFromName("Test 2009.2.03", date));
+            Assert.Equal("Test", TextUtils.StripDateFromName("Test 2009-02-3", date));
+            Assert.Equal("Test", TextUtils.StripDateFromName("Test 2009/02/3", date));
+            Assert.Equal("Test", TextUtils.StripDateFromName("Test 2009.02.3", date));
+
+            Assert.Equal("Test", TextUtils.StripDateFromName("Test Feb 3rd 09", date));
+            Assert.Equal("Test", TextUtils.StripDateFromName("Test Feb 3rd '09", date));
+            Assert.Equal("Test", TextUtils.StripDateFromName("Test Feb 3rd 2009", date));
+            Assert.Equal("Test", TextUtils.StripDateFromName("Test Feb 03 09", date));
+            Assert.Equal("Test", TextUtils.StripDateFromName("Test Feb 03 '09", date));
+            Assert.Equal("Test", TextUtils.StripDateFromName("Test Feb 03 2009", date));
+
+            // test dates in September
```

Comment:
Might be worth saying why September has extra tests (e.g. special handling of 'Sept')?